merkle_tree/src/prelude.rs

```diff
@@ -65,13 +68,19 @@ where
     S: Sponge<U = F>,
 {
     fn digest(data: &[F]) -> Result<F, MerkleTreeError> {
-        let mut input = [F::default(); INPUT_SIZE];
-        input.copy_from_slice(&data[..]);
+        let mut input = vec![F::zero(); INPUT_SIZE + 1];
+        if data.len() > INPUT_SIZE {
+            return Err(MerkleTreeError::ParametersError(
+                "input too large".to_string(),
+            ));
+        }
+        input[1..].copy_from_slice(&data[..]);
         Ok(FixedLenPoseidon2Hash::<F, S, INPUT_SIZE, 1>::evaluate(input)?[0])
     }
 
     fn digest_leaf(pos: &I, elem: &F) -> Result<F, MerkleTreeError> {
-        let mut input = [F::default(); INPUT_SIZE];
+        let mut input = vec![F::zero(); INPUT_SIZE + 1];
+        input[0] = F::one();
         input[INPUT_SIZE - 1] = F::from(pos.clone());
         input[INPUT_SIZE - 2] = *elem;
         Ok(FixedLenPoseidon2Hash::<F, S, INPUT_SIZE, 1>::evaluate(input)?[0])
```

Comment:
i don't understand. how can we not increase the input size to the hash function? why not `FixedLenPoseidon2Hash::<F, S, INPUT_SIZE + 1, 1>::evaluate()`

Reply:
"generic parameters may not be used in const operations\nconst parameters may only be used as standalone arguments, i.e. `INPUT_SIZE`"